torch/functional.py

```diff
@@ -599,3 +600,111 @@ def argsort(input, dim=None, descending=False):
     if dim is None:
         return torch.sort(input, -1, descending)[1]
     return torch.sort(input, dim, descending)[1]
+
+
+def norm(input, p="fro", dim=None, keepdim=False, out=None):
+    r"""Returns the matrix norm or vector norm of a given tensor.
+
+    Args:
+        input (Tensor): the input tensor
+        p ({int, float, inf, -inf, 'fro', 'nuc'}): the order of norm
+            The following norms can be calculated:
+            =====  ============================  ==========================
+            ord    matrix norm                   vector norm
+            =====  ============================  ==========================
+            None   Frobenius norm                2-norm
+            'fro'  Frobenius norm                --
+            'nuc'  nuclear norm                  --
+            other  as vec norm when dim is None  sum(abs(x)**ord)**(1./ord)
+            =====  ============================  ==========================
+        dim ({int, 2-tuple of ints, 2-list of ints}, optional): If it is a int,
+        vector norm will be calculated, if it is 2-tuple of ints, matrix norm
+        will be calculated. If the value is None, matrix norm will be calculated
+        when the input tensor only has two dimensions, vector norm will be
+        calculated when the input tensor only has one dimension. If the input
+        tensor has more than two dimensions, the vector norm will be applied to
+        last dimension.
+        keepdim (bool): whether the output tensors have :attr:`dim`
+            retained or not. Ignored if ``dim=None and out=None``.
+        out (Tensor, optional) – the output tensor. Ignored if
+        ``dim=None and out=None``.
+
+    Example::
+        >>> import torch
```

Comment:
Can we get rid of some of these examples? It will won't look good on the docs page. I feel like the examples should illustrate proper usage of the function, it doesn't need to cover every code path.